src/ol/Object.js

```diff
@@ -184,7 +184,11 @@ class BaseObject extends Observable {
    */
   setProperties(values, opt_silent) {
     for (const key in values) {
-      this.set(key, values[key], opt_silent);
+      if (key === 'properties') {
+        this.setProperties(values[key], opt_silent);
```

Comment:
or maybe remove this logic from setProperties and change the call in the constructor instead? I think having this in the setProperties function is not really needed.

Reply:
```
if (opt_values !== undefined) {
  this.setProperties(opt_values);
}
```
to
```

if (opt_values !== undefined) {
  if (opt_values.properties !== undefined) {
    this.setProperties(opt_values.properties);
    delete opt_values.properties;
  }
  this.setProperties(opt_values);
}
```

Reply:
I had something like that already, but was hesitant because usually we don't modify the object that gets passed to the constructor. But you convinced me that in this case it makes sense to do that, because it's just a workaround for typed environments.